pkg/cache/clusterqueue.go

```diff
@@ -518,19 +518,23 @@ func (c *clusterQueue) reportActiveWorkloads() {
 func (c *clusterQueue) updateWorkloadUsage(wi *workload.Info, m int64) {
 	admitted := workload.IsAdmitted(wi.Obj)
 	frUsage := wi.FlavorResourceUsage()
-	tasUsage := wi.TASUsage()
 	for fr, q := range frUsage {
-		tasFlvCache := c.tasFlavorCache(fr.Flavor)
 		if m == 1 {
 			addUsage(c, fr, q)
-			if tasFlvCache != nil {
-				tasFlvCache.addUsage(tasUsage)
-			}
 		}
 		if m == -1 {
 			removeUsage(c, fr, q)
-			if tasFlvCache != nil {
-				tasFlvCache.removeUsage(tasUsage)
+		}
+	}
+	if features.Enabled(features.TopologyAwareScheduling) && wi.IsUsingTAS() {
```

Comment:
`IsUsingTAS` is just a performance optimization to avoid allocation inside `TASUsage`. I could move it to a dedicated PR, OTOH it is straighforward so I'm leaving it to reviewers.

cc @PBundyra